### PR TITLE
Support plugin reload by unregistering the listener

### PR DESCRIPTION
**Pull request by rodm**
Unregistering the listener allows the plugin to be unloaded and reloaded. This allows the plugin to be deployed without restarting the TeamCity Server during development and should also allow a released version of the plugin to be upgraded without restarting the server.

### CODE REVIEW

File: src/main/java/nu/studer/teamcity/buildscan/BuildScanBuildServerListener.java [hunk 1]
```diff
@@ -39,6 +39,11 @@ public void register() {
         LOGGER.info(String.format("Registered %s. %s-%s", getClass().getSimpleName(), pluginDescriptor.getPluginName(), pluginDescriptor.getPluginVersion()));
     }
 
+    public void unregister() {
+        buildServer.removeListener(this);
+        LOGGER.info(String.format("Unregistered %s. %s-%s", getClass().getSimpleName(), pluginDescriptor.getPluginName(), pluginDescriptor.getPluginVersion()));
+    }
+
```
Comment (etiennestuder) [inline src/main/java/nu/studer/teamcity/buildscan/BuildScanBuildServerListener.java:46]: I believe there are more beans to unregister than this one. I will take a look.

**Comment by etiennestuder** [comment]
@rodm Thanks. Just curious:
- How do you trigger the unloading and reloading of a plugin?
- How do you upgrade to a newer version of a TC plugin without restart?

**Comment by etiennestuder** [comment]
Thanks, @rodm. I will include this in v0.21.

**Comment by rodm** [comment]
@etiennestuder 
> @rodm Thanks. Just curious:
> 
>     * How do you trigger the unloading and reloading of a plugin?
To reload with Gradle plugin there is a `deployTo` task or via the UI on the plugins page
> 
>     * How do you upgrade to a newer version of a TC plugin without restart?
On the plugins page there is a 'Check now' button, this will list any updates and allow them to be installed, if they cannot be reloaded a warning is shown that a server restart is required.

**Comment by etiennestuder** [comment]
Thanks for the details.

**Comment by etiennestuder** [comment]
I've released v0.21 with these changes.

**Pull request closed without merging**